### ISSUE TITLE
plugin generates invalid type: date schema

### ISSUE THREAD

**Issue by m-shaka** [nestjs/swagger#836]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

Swagger plugin generates invalid `type: date` schema as Date paramter of query.

## Expected behavior

Expceted below.

```yaml
schema:
  type: string
  format: date
```

## Minimal reproduction of the problem with instructions

https://github.com/m-shaka/nest-swagger-query

Please clone the repository, run `npm i` and `npm run start:dev`, and then open http://localhost:3000/api-json .
I generated initial codes with nest cli, and then changed three files.
```TypeScript
// app.dto.ts
export class MyQueryParam {
    fromDate: Date;
}

// app.controller.ts
import { Controller, Get, Query } from '@nestjs/common';
import { AppService } from './app.service';
import { MyQueryParam } from './app.dto'

@Controller()
export class AppController {
  constructor(private readonly appService: AppService) { }

  @Get()
  getHello(@Query('q') q: Date): string {
    return this.appService.getHello();
  }

  @Get('/class')
  classQuery(@Query() q: MyQueryParam): string {
    return this.appService.getHello();
  }
}

// main.ts
import { NestFactory } from '@nestjs/core';
import { SwaggerModule, DocumentBuilder } from '@nestjs/swagger';
import { AppModule } from './app.module';

async function bootstrap() {
  const app = await NestFactory.create(AppModule);

  const options = new DocumentBuilder()
    .setTitle('Cats example')
    .setDescription('The cats API description')
    .setVersion('1.0')
    .build();
  const document = SwaggerModule.createDocument(app, options);
  SwaggerModule.setup('api', app, document);

  await app.listen(3000);
}
bootstrap();
```

Output

```yaml
parameters:
  - name: fromDate
    required: true
    in: query
    schema:
       type: date
```

## Environment

<pre><code>
Nest version:
├── @nestjs/cli@7.4.1
├── @nestjs/common@7.3.1
├── @nestjs/core@7.3.1
├── @nestjs/platform-express@7.3.1
├── @nestjs/schematics@7.0.1
├── @nestjs/swagger@4.5.12
├── @nestjs/testing@7.3.1

 
For Tooling issues:
- Node version: v10.13.0  
- Platform: Ubuntu 18.04 

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Can you please provide a minimum reproduction repository? This would help us to solve this issue quicker

**Comment by m-shaka** [comment]
So sorry. I've updated the first comment.

Please clone the [repo](https://github.com/m-shaka/nest-swagger-query).

**Comment by kamilmysliwiec** [comment]
This should be fixed in 4.7.3
